aspnet/mvc/overview.rst

```diff
@@ -1,8 +1,288 @@
+
+Web APIs
+^^^^^^^^
+
+In addition to being a great platform for building web sites, ASP.NET Core MVC has great support for building Web APIs. You can build services that can reach a broad range of clients including browsers and mobile devices.
+
+The framework includes support for HTTP content-negotiation with built-in support for :doc:`formatting data </mvc/models/formatting>` as JSON or XML. Write :doc:`custom formatters </mvc/models/custom-formatters>` to add support for your own formats. 
+
+Use link generation to enable support for hypermedia. Easily enable support for `cross-origin resource sharing (CORS) <http://www.w3.org/TR/cors/>`__ so that your Web APIs shared across multiple Web applications.
+
+Testability
+^^^^^^^^^^^
+
+The framework's use of interfaces and dependency injection make it well-suited to unit testing, and the framework includes features (like a TestHost and InMemory provider for Entity Framework) that make :doc:`integration testing </testing/integration-testing>` quick and easy as well. Learn more about :doc:`testing controller logic </mvc/controllers/testing>`.
+
+Razor view engine
+^^^^^^^^^^^^^^^^^
+
+:doc:`ASP.NET Core MVC views </mvc/views/overview>` use the the :doc:`Razor view engine </mvc/views/razor>` to render views. Razor is a compact, expressive and fluid template markup language for defining views using embedded .NET code. Razor is used to dynamically generate web content on the server. Unlike most template syntaxes, you do not need to interrupt your coding to explicitly denote server blocks within your HTML. The parser is smart enough to infer this from your code. This makes creating views in Razor clean, fast and fun!
+
+.. code-block:: text
+
+  <ul>
+    @for (int i = 0; i < 5; i++) {
+      <li>List item @i</li>
+    }
+  </ul>
+
+Using the Razor view engine you can define :doc:`layouts </mvc/views/layout>`, :doc:`partial views </mvc/views/partial>` and replaceable sections.
+
+Strongly typed views
+^^^^^^^^^^^^^^^^^^^^
+
+Razor views in MVC can be strongly typed based on your model. A strongly-typed view explicitly specifies the model type for the view. You can then leverage the compiler and IntelliSense to ensure that references to property names and types in your views are correct:
+
+.. code-block:: html
+
+  @model IEnumerable<Product>
+  <ul> 
+      @foreach (Product p in Model) 
+      { 
+          <li>@p.Name</li>
+      }
+  </ul>
+
+Tag Helpers
+^^^^^^^^^^^
+
+:doc:`Tag Helpers </mvc/views/tag-helpers/intro>` enable you to dynamically generate or modify the HTML returned to the client. You can use tag helpers to define custom tags or to modify the behavior of existing tags. Tag Helpers bind to specific elements based on the element name and its attributes. Tag Helpers give you all of the benefits of server-side rendering while still preserving an HTML editing experience.
```

Comment:
@NTaylorMullen  we need an elevator pitch for TH in this "MVC Overview" targeted at new folks considering the move to MVC Core.

@danroth27  @spboyer   please comment.

I like my intro better, altho a few customers told me they don't like the image TH.

Consider:

ag Helpers enable server

-- side code to participate in creating and rendering HTML elements in Razor files. For example, the built-in `ImageTagHelper <https://docs.asp.net/projects/api/en/latest/autoapi/Microsoft/AspNetCore/Mvc/TagHelpers/ImageTagHelper/index.html>`__ can append a version number to the image name. Whenever the image changes, the server generates a new unique version for the image, so clients are guaranteed to get the current image (instead of a stale cached image). There are many built-in Tag Helpers for common tasks - such as creating forms, links, loading assets and more - and even more available in public GitHub repositories and as NuGet packages.
Tag Helpers are authored in C#, and they target HTML elements based on element name, attribute name, or parent tag. For example, the built-in ``LinkTagHelper`` - show your example

TH Provide an HTML-friendly development experience and a rich IntelliSense environment for creating HTML and Razor markup. Most of the built-in Tag Helpers target existing HTML elements and provide server-side attributes for the element.